File: src/main/java/org/genyris/format/AbstractFormatFunction.java
package org.genyris.format;

import org.genyris.exception.GenyrisException;
import org.genyris.interp.ApplicableFunction;
import org.genyris.interp.Interpreter;

public abstract class AbstractFormatFunction extends ApplicableFunction {

    public AbstractFormatFunction(Interpreter interp, String name, boolean eager) {
        super(interp, name, eager);
    }

    public static void bindFunctionsAndMethods(Interpreter interpreter) throws GenyrisException {
        interpreter.bindGlobalProcedureInstance(new PrintFunction(interpreter));
        interpreter.bindGlobalProcedureInstance(new WriteFunction(interpreter));
        interpreter.bindGlobalProcedureInstance(new DisplayFunction(interpreter));
    }


}

File: src/org/genyris/format/AbstractFormatFunction.kt
package org.genyris.format

import org.genyris.exception.GenyrisException
import org.genyris.interp.ApplicableFunction
import org.genyris.interp.Interpreter

abstract class AbstractFormatFunction(interp: Interpreter, name: String?, eager: Boolean) :
    ApplicableFunction(interp, name, eager) {
    companion object {
        @kotlin.Throws(GenyrisException::class)
        fun bindFunctionsAndMethods(interpreter: Interpreter) {
            interpreter.bindGlobalProcedureInstance(PrintFunction(interpreter))
            interpreter.bindGlobalProcedureInstance(WriteFunction(interpreter))
            interpreter.bindGlobalProcedureInstance(DisplayFunction(interpreter))
        }
    }
}